### PR TITLE
TE-301 Icon element

### PR DESCRIPTION
**Pull request by marcdavi-es**
[Related YouTrack issue](https://youtrack.lodgify.net/issue/TE-301)

### What **one** thing does this PR do?
Implements the Icon element using SVG.

icons

### CODE REVIEW

File: src/components/elements/Icon/Readme.md [hunk 1]
```diff
@@ -2,6 +2,19 @@
...
+### Content
+
+An icon can display a named svg from the Lodgify UI icon library or any valid svg path description.
+
+```jsx
+<Icon label="Using props.name" name="search" />
+<Divider />
+<Icon
+  label="Using props.path"
+  path="M9.67,13.55V12.49c0-2.11,3.48-3,3.48-4.66a1.73,1.73,0,0,0-1.89-1.48c-1.76,0-2.24,1.58-2.87,1L6.85,5.87a.45.45,0,0,1,0-.65A7.07,7.07,0,0,1,12,3c2.66,0,5.13,1.86,5.13,4.41,0,3.32-3.77,3.42-3.77,6.14a.48.48,0,0,1-.48.48h-2.7A.48.48,0,0,1,9.67,13.55Zm1.82,2.79a2.45,2.45,0,0,1,2.41,2.42,2.39,2.39,0,1,1-2.41-2.42Z"
```
Comment (Jonnygwi) [inline src/components/elements/Icon/Readme.md]: 🚀

File: src/components/elements/Icon/component.spec.js [hunk 2]
```diff
@@ -1,52 +1,85 @@
...
-    it('should render a single Lodgify UI `Paragraph` component', () => {
-      const label = '🐘';
-      const wrapper = getIcon({ label });
-      const actual = wrapper.find(Paragraph);
-      expect(actual).toHaveLength(1);
+  describe('the `path` element', () => {
+    it('should get the right props', () => {
+      const wrapper = getIcon().find('path');
+      expectComponentToHaveProps(wrapper, {
+        d: expect.any(String),
```
Comment (Jonnygwi) [inline src/components/elements/Icon/component.spec.js]: d == grim

Comment (Jonnygwi) [reply src/components/elements/Icon/component.spec.js]: Not sure if we should worry about that though

File: src/components/elements/Icon/component.js [hunk 3]
```diff
@@ -1,40 +1,96 @@
...
+    className={getClassNames('icon', color, size, {
+      circular: isCircular,
+      'inverted grey': isDisabled,
+      inverted: isColorInverted,
+    })}
+    // Passing through props is required for
+    // compatibility with dynamic components e.g. `ToolTip`
+    {...otherProps}
+  >
+    <svg viewBox="0 0 24 24">
```
Comment (davidpelayo) [inline src/components/elements/Icon/component.js]: Doesn't the viewBox size need to correlate with `props.size`?

File: src/semantic/src/themes/livingstone/elements/icon.overrides [hunk 4]
```diff
@@ -20,14 +20,32 @@
 
 /* Icon Variables */
 
+i.icon {
+  height: @height;
+  width: auto;
+
+  font-family: inherit;
+  vertical-align: top;
```
Comment (davidpelayo) [inline src/semantic/src/themes/livingstone/elements/icon.overrides]: Ideally this should be `middle`.

File: src/components/elements/Icon/component.js (hunk 3 above)
Comment (marcdavi-es) [reply src/components/elements/Icon/component.js]: Good question. 

We need `viewBox` static at `0 0 24 24` (or whatever) to make sure the `path` scales correctly. 

The size of the `svg` as a whole is define by its width and height, which are in turn proportionate to the `font-size` of the containing `i` element.

File: src/semantic/src/themes/livingstone/elements/icon.overrides (hunk 4 above)
Comment (marcdavi-es) [reply src/semantic/src/themes/livingstone/elements/icon.overrides]: This makes sure the containing `i` respects the `line-height` of the surrounding context.

**Pull request merged**